include/deal.II/matrix_free/operators.h

```diff
@@ -160,6 +160,12 @@ namespace MatrixFreeOperators
     virtual void compute_diagonal () = 0;
 
     /**
+     * Get read access to the MatrixFree object stored with this operator.
+     */
+    const std_cxx11::shared_ptr<const MatrixFree<dim,Number> > &
```

Comment:
@drwells I agree in this particular context. My motivation for returning the shared pointer was that one can construct an additional operator with the given reference, say a mass matrix with the `shared_ptr` return from a Laplace matrix. Sometimes the two are close in user code, but sometimes they appear in different contexts and then the ability to get the matrix-free object (without storing it in the original context) comes handy.